### PR TITLE
DiagnosticVerifier: Support line offsets in fix-it verification ranges

### PR DESCRIPTION
**Pull request by AnthonyLatsis**
Follow-up to #39572

The range syntax is now `([+-]?N:)?N-([+-]?N:)?N` where `N` is `[0-9]+`.

### CODE REVIEW

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

**Review by rintaro — COMMENTED** [review]
Thank you! Just a few comments about code styling. Implementation looks pretty good

File: lib/Frontend/DiagnosticVerifier.cpp [hunk 1]
```diff
@@ -339,6 +339,115 @@ DiagnosticVerifier::renderFixits(ArrayRef<CapturedFixItInfo> ActualFixIts,
...
+
+      switch (Str.front()) {
+      case '+':
+        return LineOffsetKind::Plus;
+      case '-':
+        return LineOffsetKind::Minus;
+      default:
+        return LineOffsetKind::None;
+      }
+    }();
```
[the remark this answers is not in the document]
Comment (AnthonyLatsis) [reply lib/Frontend/DiagnosticVerifier.cpp:372]: I knew this was gonna bug someone 😄

No problem, can I keep the lambda for constness though? When reading C++, I feel a ton of relief every time I see that something's not going to be mutated in the next 50+ lines.

[the remark this answers is not in the document]
Comment (AnthonyLatsis) [reply lib/Frontend/DiagnosticVerifier.cpp:372]: Actually, never mind; the difference looks negligible this time.

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by rintaro — APPROVED** [review]

**Comment by AnthonyLatsis** [comment]
@rintaro Appreciate the review!

**Pull request merged**